Training/Training/Training.c

```diff
@@ -1,14 +1,161 @@
-// Program.c
-// Program on main branch.
-// ------------------------------------------------------------------------------------------------
-
-#include<stdio.h>
-
-int main () {
-	printf ("Hello World!");
-	return 0;
+// ------------------------------------------------------------------------------------------------
+// Training ~ A training program for new joiners at Metamation, Batch - July 2024.
+// Copyright (c) Metamation India.
+// ------------------------------------------------------------------
+// Training.c
+// Program to check for palindrome of a string or an integer
+// Sudarson S
+// ------------------------------------------------------------------------------------------------
+
+#include <ctype.h>
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <stdbool.h>
+
+// Input limit is 200 characters
+#define MAX 202
+#define PALINDROME 1
+#define NOTPALINDROME 0
+#define INVALID -1
+#define ANSI_COLOR_RED     "\x1b[31m"
+#define ANSI_COLOR_GREEN   "\x1b[32m"
+#define ANSI_COLOR_YELLOW  "\x1b[33m"
+#define ANSI_COLOR_RESET   "\x1b[0m"
+
+/// <summary>Function which checks if a given string is palindrome or not</summary>
+int PalindromeChecker (char* phrase) {
+   int i = 0, validChar = 0, length = (int)strlen (phrase), j = length - 1;
+   if (length > MAX - 2)
+      return INVALID;
+   // Checking if valid characters from the beginning and the end match with each other
+   while (i <= j) {
+      if (isalnum (phrase[i]) == 0)
+         i++;
+      else if (isalnum (phrase[j]) == 0)
+         j--;
+      else
+         if (toupper (phrase[i]) != toupper (phrase[j]))
+            return NOTPALINDROME;
+         else
+            validChar++, i++, j--;
+   }
+   return validChar != 0 ? PALINDROME : INVALID;
+}
+
+/// <summary>Function which checks if the input number is palindrome or not</summary>
+long long int NumberReverser (int number) {
+   int numTemp;
+   long long int rev = 0;
+   if (number >= 0 && number <= 9) {
+      rev = number;
+      return rev;
+   }
+   numTemp = number;
+   while (numTemp != 0) {
+      rev = rev * 10 + numTemp % 10;
+      numTemp /= 10;
+   }
+   return rev;
+}
+
+/// <summary>Function to drain the unassigned inputs from stdin</summary>
+void BufferDrain () {
+   while (getchar () != '\n');
+}
+
+/// <summary>Function to check if the given input is a phrase or a number</summary>
+bool IsValidNum (char* phrase, int* num) {
+   errno = 0;
+   char* endptr = NULL;
+   *num = strtol (phrase, &endptr, 10);
+   return endptr == phrase || *endptr != '\0' || errno == ERANGE || phrase[0] == ' ' || phrase[0] == '\t' ?
+      false : true;
+}
+
+/// <summary>Function to run test cases</summary>
+void TestCases () {
+   int Length;
+   char* Inputs[] = { "Was it a car or a cat I saw?", "I did, did I?", "Don't nod", "TRUMPF",
+      "Metamation", "121", "232", "123454321", "123456789", "0", "!@#  $%", " ", "-878" }, * Expected[] = {

```

Comment:
```suggestion
      "Metamation", "121", "232", "123454321", "123456789", "0", "!@#  $%", " ", "-878" },
       * expected[] = {

```
For readability, this can go on the next line, while still in the same statement. In C, a statement only ends with a ';'.